R/import_ebird.R

```diff
@@ -80,8 +146,60 @@ ebird_col_type <- function(col_names) {
   # assume anything else is character
   col_types <- col_types[col_names]
   col_types[is.na(col_types)] <- "string"
-  setNames(col_types, col_names)
+  names(col_types) <- col_names
+  return(col_types)
 }
 
-# Download URLs, eg https://download.ebird.org/ebd/prepackaged/ebd_relJul-2021.tar
-# Note, may take > 24hrs to complete
+extract_metadata <- function(tarfile) {
+  stopifnot(is.character(tarfile), length(tarfile) == 1, file.exists(tarfile))
+  
+  f <- basename(tarfile)
+  # checks for validity of filename
+  if (!grepl("\\.tar$", f)) {
+    stop("The provided file does not appear to be a tar archive. The file ",
+         "extension should be .tar.")
+  }
+  if (grepl("ebd_sampling_rel[A-Z]{1}[a-z]{2}-[0-9]{4}\\.tar$", f)) {
```

Comment:
these `if (grepl(...` commands maybe handy elsewhere, and this code would probably be easier to read if they were wrapped out into helper functions, eg. as `is_checklist()` or `is_observation()`